modules/develop/pages/produce-data/configure-producers.adoc

```diff
@@ -232,6 +232,24 @@ verbose format, which impacts your compression speeds and network throughput.
 Alternatively, going with AVRO or Protobuf allows you to only define the schema
 in one place, while also enabling features like schema evolution.
 
+[[broker_timestamps]]
+=== Broker timestamps
+
+Redpanda employs a unique strategy to help optimize retention. Every time a producer posts a message to a topic, Redpanda records the broker's system date and time in the message's `broker_timestamp` property. This helps ensure retention functions as intended in cases where a producer submits a message with a created timestamp skewed from the broker's time. Closed segments are only eligible for deletion under Redpanda's retention policy when the age of all messages in the segment exceed a set threshold. If the message's timestamp is in the future it will never exceed this policy and the segment is not eligible for deletion under Redpanda's cleanup processing.
```

Comment:
I reworded it -- I was trying to explain the problem this was solving and what makes it particularly unique to Redpanda.
 
"Closed segments are only eligible for deletion under Redpanda's retention policy when the age of all messages in the segment exceed a set threshold. Using broker timestamps ensures that skewed messages are cleaned up appropriately where they may otherwise appear to never exceed the retention policy threshold."